### PR TITLE
feat: Add new camera device services

### PR DESCRIPTION
**Pull request by lenny-intel**
Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) ** N/A**
  Documented in README

## Testing Instructions
TBD

### CODE REVIEW

**Review by jim-wang-intel — DISMISSED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@weichou1229 @FelixTing please help review the related camera parts

**Review by weichou1229 — APPROVED** [review]
LGTM

File: compose-builder/add-device-usb-camera.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - data
+      - metadata
+    security_opt:
+      - no-new-privileges:true
+    user: root:root
+    device_cgroup_rules:
+      - 'c 81:* rw' # c:character device 81:device major number(81=webcam) *:device minor number rw:read/write
+    volumes:
+      - /dev:/dev
+      - /run/udev:/run/udev:ro
```
Comment (FelixTing) [inline compose-builder/add-device-usb-camera.yml:47]: add a newline at the end of file

Comment (lenny-intel) [reply compose-builder/add-device-usb-camera.yml]: Done

**Comment by lenny-intel** [comment]
@ajcasagrande , should the generated compose file have the following override placeholders for device-onvif-camera?

```
     DRIVER_DISCOVERYSUBNETS: "<your subnets>"
      DRIVER_SCANPORTS: "<your scan ports>"
      DRIVER_DISCOVERYETHERNETINTERFACE: "<your interface name>
      DRIVER_DEFAULTSECRETPATH: "<your default secret path>
```
Here are the values used by validation:
```
      DRIVER_DISCOVERYSUBNETS: "192.168.4.0/24,192.168.0.0/24"
      DRIVER_SCANPORTS: "80,2020"
      DRIVER_DISCOVERYETHERNETINTERFACE: enp0s31f6
      DRIVER_DEFAULTSECRETPATH: credentials001
```
I am assume all these will be the be tweaked via overrides anytime the compose file is generated, correct?
What about ScanPorts? can/should "80,2020" be the default? Why is 80 being scanned?

**Comment by ajcasagrande** [comment]
> @ajcasagrande , should the generated compose file have the following override placeholders for device-onvif-camera?
> 
> ```
>      DRIVER_DISCOVERYSUBNETS: "<your subnets>"
>       DRIVER_SCANPORTS: "<your scan ports>"
>       DRIVER_DISCOVERYETHERNETINTERFACE: "<your interface name>
>       DRIVER_DEFAULTSECRETPATH: "<your default secret path>
> ```
> 
> Here are the values used by validation:
> 
> ```
>       DRIVER_DISCOVERYSUBNETS: "192.168.4.0/24,192.168.0.0/24"
>       DRIVER_SCANPORTS: "80,2020"
>       DRIVER_DISCOVERYETHERNETINTERFACE: enp0s31f6
>       DRIVER_DEFAULTSECRETPATH: credentials001
> ```
> 
> I am assume all these will be the be tweaked via overrides anytime the compose file is generated, correct? What about ScanPorts? can/should "80,2020" be the default? Why is 80 being scanned?

`DRIVER_SCANPORTS` doesn't even exist anymore (technically never did since its still WIP PR :wink: ). It is a constant port `3702` as we are no longer probing for ONVIF but instead simulating ws-discovery on a per-device basis.

the other ones can be left out as well, unless you just want the user to know they have the option of changing them?

**Comment by lenny-intel** [comment]
> the other ones can be left out as well, unless you just want the user to know they have the option of changing them?

Yea, that is what I was thinking since they will have to change them, correct?

**Comment by lenny-intel** [comment]
@ajcasagrande , I think I will not add those placeholder overrides. The User Docs should specify adding them or using Consul to change the values.

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Comment by ajcasagrande** [comment]
> @ajcasagrande , I think I will not add those placeholder overrides. The User Docs should specify adding them or using Consul to change the values.

Yeah I think that makes the most sense.

File: compose-builder/README.md [hunk 2]
```diff
@@ -52,8 +52,12 @@ This folder contains the following compose files:
     Secure delayed start services **extending** compose file. Adds additional delayed start services based on spire/spiffe implementation to provide the secret store token on the runtime secure configuration.
 - **add-device-bacnet.yml**
     Device Service **extending** compose file, which adds the **Device Bacnet**  service.
-- **add-device-camera.yml**
-    Device Service **extending** compose file, which adds the **Device Camera**  service.
+- **add-device-camera.yml** (***not this service will be Deprecated, use Device ONVIF Camera***)
+    Device Service *extending** compose file, which adds the **Device Camera**  service.
```
Comment (ajcasagrande) [inline compose-builder/README.md:56]: Missing a `*`?

File: compose-builder/README.md [hunk 3]
```diff
@@ -52,8 +52,12 @@ This folder contains the following compose files:
     Secure delayed start services **extending** compose file. Adds additional delayed start services based on spire/spiffe implementation to provide the secret store token on the runtime secure configuration.
 - **add-device-bacnet.yml**
     Device Service **extending** compose file, which adds the **Device Bacnet**  service.
-- **add-device-camera.yml**
-    Device Service **extending** compose file, which adds the **Device Camera**  service.
+- **add-device-camera.yml** (***not this service will be Deprecated, use Device ONVIF Camera***)
```
Comment (ajcasagrande) [inline compose-builder/README.md:55]:
```suggestion
- **add-device-camera.yml** (***note that this service will be Deprecated, use Device ONVIF Camera***)
```

Comment (lenny-intel) [reply compose-builder/README.md]: Fixed

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**